design/KEPS/namespace-discovery/proposal.md

```diff
@@ -12,19 +12,162 @@ Provide a better way for namespace discovery within both server and operator.
 
 # Motivation
 
-Currently, the Kiali Server and Operator have many different ways to configure which namespaces should be accessible and viewable by the Kiali UI. The configurations are complex and confusing to users. We want to make things more easily configurable and understandable while still maintaining the ability for Kiali to have either cluster-wide access or limited access to specific namespaces.
+Currently, the Kiali Server and Operator have many different ways to configure which namespaces should be accessible and viewable by the Kiali UI. The configurations are complex and confusing to users. We want to make things more easily configurable and understandable while still maintaining the ability for Kiali to have either cluster-wide access or limited access to specific namespaces. We also want Kiali to support soft multitenancy using the same mechanism as Istio to make configuration more consistent with Istio and easier for users.
 
 ## Goals
 
 - Provide an easy way for users to configure Kiali UI to have access to specific namespaces.
 - Provide a way for Kiali Server to have either cluster-wide access to all namespaces or limited access to a subset of namespaces.
 - Remove the deprecated functionality that provided the old way of filtering and limiting namespace access.
+- Enable soft multi-tenancy Kiali deployments where multiple Kialis with cluster wide access can be deployed to the same cluster and scoped to specific namespaces.
 
 # Solution
 
-Provide a new list of "discovery selectors" in the same way that Istio itself has discovery selectors. In fact, the solution should include Kiali Server and/or Operator using the Istio Discovery Selectors where appropriate, and using Kiali's own discovery selectors as an optional mechanism to override Istio discovery selectors.
+Provide a new list of "discovery selectors" in the same way that Istio itself has discovery selectors. Kiali will re-use the Istio Discovery Selectors it retreive's from the control plane's `istio` configmap. Kiali will also have a new config option `deployment.discovery_selectors` that, when specified, will be used instead of the control plane's Istio Discovery Selectors.
```

Comment:
```suggestion
Provide a new list of "discovery selectors" in the same way that Istio itself has discovery selectors. Kiali will re-use the Istio Discovery Selectors it retreives from the control plane's `istio` configmap. Kiali will also have a new config option `deployment.discovery_selectors` that, when specified, will be used instead of the control plane's Istio Discovery Selectors.
```